src/helpers/SuggestionRegistry.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License", destination); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export type Suggestion = {
+	value: string;
+	description?: string;
+	group?: string;
+};
+
+export type SuggestionContext = {
+	propertyName: string;
+	inputValue: string | number;
+	cursorPosition?: number | null;
+};
+
+export type SuggestionProviderFunction = (word: string, context: SuggestionContext) => Suggestion[] | Promise<Suggestion[]>;
+
+const suggestionRegistry = new Map<string, SuggestionProviderFunction>();
+
+/**
+ * Registers a new suggestion provider for a given topic
+ */
+export function registerSuggestionProvider(topic: string, providerFn: SuggestionProviderFunction): void {
+	suggestionRegistry.set(topic, providerFn);
+}
```

Comment:
[friendly-comment] Not strictly necessary right now, but if in the future we need to clear this registry, we can't.

I can see 2 scenarios where we might want to unregister a provider, one when there's a disconnection from one of the sources, and maybe for testing purposes.

If we reach one of those scenarios, we could integrate the `suggestionRegistry` and the `registrySuggestionProvider` function into a class so we could offer an `unregister` method as well.